cmd/action/internal/update/update.go

```diff
@@ -0,0 +1,152 @@
+package update
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/mitchellh/go-homedir"
+	"github.com/pelletier/go-toml/v2"
+	"golang.org/x/mod/semver"
+)
+
+// Update logic:
```

Comment:
1. Fix the documentation text (more than 24h, hcl)
2. Put this is a Godoc comment on the relevant funcs